train.py

```diff
@@ -364,6 +371,47 @@ def loss_fn(pred, labels):
                 time_last_log = time.perf_counter()
                 gpu_memory_monitor.reset_peak_stats()
 
+            # log val metrics
+            if (
+                job_config.validation.enable_val
+                and train_state.step == 1
+                or train_state.step % job_config.validation.eval_freq == 0
+            ):
+                with torch.no_grad():
+                    model.eval()
+                    val_data_loader = build_hf_data_loader(
+                        job_config.validation.dataset,
+                        job_config.validation.dataset_path,
+                        tokenizer,
+                        val_bs,
+                        job_config.training.seq_len,
+                        dp_degree,
+                        dp_rank,
+                        False
+                    )
+                    num_flop_per_token_val = utils.get_num_flop_per_token_forward(
+                        utils.get_num_params(model, exclude_embedding=True),
+                        model_config,
+                        job_config.training.seq_len,
+                    )
+                    validate(
+                        model,
+                        val_data_loader,
+                        eval_state,
+                        logger,
```

Comment:
Passing the logger isn't necessary, simply initialize the logger with the init_logger function in val.py